src/entity/bookType/BookType.java

```diff
@@ -0,0 +1,96 @@
+package entity.bookType;
+
+import entity.category.Category;
+import entity.category.Subcategory;
+import entity.review.Review;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+
+public abstract class BookType{
+    protected int id;
+    protected String name, author, description, availability;
```

Comment:
even if youu have a lot of Strings, put them on sepparate lines for visibility, best practice in a poject